internal/fs/fs.go

```diff
@@ -2769,6 +2828,18 @@ func (fs *fileSystem) ReadFile(
 		op.Dst, op.BytesRead, err = fh.Read(ctx, op.Dst, op.Offset, fs.sequentialReadSizeMb)
 	}
 
+	// A FileClobberedError indicates the underlying GCS object has changed,
+	// making the kernel's dentry for this file stale. We use the notifier to
+	// invalidate this entry, providing feedback to the kernel about the dynamic
+	// content change and ensuring subsequent lookups fetch the correct metadata.
+	if err != nil && fs.newConfig.FileSystem.ExperimentalEnableDentryCache {
```

Comment:
nit: can we move the error check inside 

if DentryCache {
    if err != nil && errors.Is(clobbered) {
       invalidate()
  }
}